watcher/local/watcher_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	ethChannel "perun.network/go-perun/backend/ethereum/channel"
+	_ "perun.network/go-perun/backend/ethereum/channel/test" // For initilizing channeltest
+	"perun.network/go-perun/channel"
+	channeltest "perun.network/go-perun/channel/test"
+	"perun.network/go-perun/pkg/test"
+	"perun.network/go-perun/watcher/internal/mocks"
+	"perun.network/go-perun/watcher/local"
+)
+
+func Test_StartWatching(t *testing.T) {
+	rng := test.Prng(t)
+	rs := &mocks.RegisterSubscriber{}
+	rs.On("Subscribe", mock.Anything, mock.Anything).Return(&ethChannel.RegisteredSub{}, nil)
+
+	t.Run("happy/ledger_channel", func(t *testing.T) {
+		w := newWatcher(t, rs)
+		params, state := channeltest.NewRandomParamsAndState(rng, channeltest.WithVersion(0))
+		signedState := makeSignedStateWDummySigs(params, state)
+
+		statesPub, eventsSub, err := w.StartWatchingLedgerChannel(context.TODO(), signedState)
+
+		require.NoError(t, err)
+		require.NotNil(t, statesPub)
+		require.NotNil(t, eventsSub)
+
+		_, _, err = w.StartWatchingLedgerChannel(context.TODO(), signedState)
+		require.Error(t, err, "StartWatching twice for the same channel must error")
+	})
+
+	t.Run("happy/sub_channel", func(t *testing.T) {
+		w := newWatcher(t, rs)
+
+		// Start watching for ledger channel.
+		parentParams, parentState := channeltest.NewRandomParamsAndState(rng, channeltest.WithVersion(0))
+		parentSignedState := makeSignedStateWDummySigs(parentParams, parentState)
+		statesPub, eventsSub, err := w.StartWatchingLedgerChannel(context.TODO(), parentSignedState)
```

Comment:
avoid context.TODO

Reply:
Updated in 67d3768492dde489a2e281f21129382504dc80f3